### ISSUE TITLE
How is accuracy measured?

### ISSUE THREAD

**Issue by Robin2091** [zzh8829/yolov3-tf2#132]
Hello,

I was wondering how the accuracy of the model is measured and how it differs from mAP. I am training with yolov3 tiny and I get only a 30 percent accuracy for "yolo_0_output". 

I fixed my dataset, got more data and then I was able to get a 45 percent accuracy. However, throughout training, the accuracy value just fluctuated between 40-47 percent and I didn't see any significant improvement in the accuracy. I then retrained again, and I was back to a 30 percent accuracy for some reason. So I don't understand why my accuracy value is fluctuating so much and why I am getting different accuracies when I trained on 2 separate occasions with the same dataset. Also, I measured the mAP value using another repo and despite the differences in accuracy, I still get around the same mAP value of 65-68 percent. 

If someone can shed some light on how accuracy is measured and why I am seeing fluctuating accuracies it would be really helpful. 

Thank you

**Comment by zzh8829** [comment]
Did you use transfer learning? how are you measuring the accuracy?

**Comment by Robin2091** [comment]
@zzh8829 Yes, I used transfer learning. On the modile. compile function i put metrics = ['accuracy']. I also tested the mAP with a different repo and I got an AP of 70 percent with an iou of 0.3.

**Comment by asquare92** [comment]
> @zzh8829 Yes, I used transfer learning(darknet). On the modile. compile function i put metrics = ['accuracy']. I also tested the mAP with a different repo and I got an AP of 70 percent with an iou of 0.3.
> my code:
> ![image](https://user-images.githubusercontent.com/57192713/71524914-63009c80-2884-11ea-9825-3a9f29474668.png)

Hi,
Can you please guide how did you calculate the mAP using the code from this repo? From your comment above, I can't see any `metrics = ['accuracy'] `parameter passed to a function. And furthermore, can you also share from which other repo did you calculate the mAP?

**Comment by Robin2091** [comment]
@asquare92  I don't believe the accuracy metric gives a good measure of the actual performance of the model. This is because Nonmax suppression is not applied during training. Also, I am not sure that the built-in accuracy metric does the same calculations as the mAP. So I used a separate repo. 

https://github.com/rafaelpadilla/Object-Detection-Metrics

**Comment by asquare92** [comment]
> @asquare92 I don't believe the accuracy metric gives a good measure of the actual performance of the model. This is because Nonmax suppression is not applied during training. Also, I am not sure that the built-in accuracy metric does the same calculations as mAP. So I used a separate repo.
> 
> https://github.com/rafaelpadilla/Object-Detection-Metrics

@Robin2091 Thanks.
Just a small question: How did you make the **detection files** after training the model using this repo? 
And for the ground truth files, I just need to use my annotations?

**Comment by Robin2091** [comment]
@asquare92 I loaded with the trained weights. I just looped through a set of images and ran the detection on the images. I saved the bounding box information in a text file. Yes, use your annotations as ground truth files.

**Comment by asquare92** [comment]
> @asquare92 I loaded the model with the trained weights. I just looped through a set of images and ran the detection on the images. I saved the bounding box information in a text file. Yes, use your annotations as ground truth files.

Is it possible for you to share the script that you wrote to create detection files?
